flowpipe/graph.py

```diff
@@ -468,6 +468,40 @@ def node_repr(self):
                 canvas_.add_item(item_)
             x += x_diff
 
+        # Include the input groups if any have been set
+        y_off = 2
+        locked_items = []
+        if self.input_groups:
+            for input_group in self.input_groups.values():
+                y_off += 1
+                i = item.Item("o {0}".format(input_group.name), [0, y_off])
+                canvas_.add_item(i)
+                locked_items.append(i)
+                for p in input_group.plugs:
+                    y_off += 1
+                    i = item.Item("`-{0}.{1}".format(p.node.name, p.name), [2, y_off])
+                    canvas_.add_item(i)
+                    locked_items.append(i)
+
+        # Move all items down by Y
+        for i in canvas_.items:
+            if i not in locked_items:
+                i.position[0] += 2
+                i.position[1] += y_off + 1 + int(bool(self.input_groups))
+
+        canvas_.add_item(item.Rectangle(x, canvas_.bbox[3] + 1, [0, 0]), 0)
+
+        # Crop the name of the graph if it is too long
+        name = self.name
+        if len(name) > x - 2:
```

Comment:
At least in py3, this conditional is not necessary, slices bounds can extend beyond the actual length of the object being sliced.

Reply:
Cool, I didn't know that. Will still keep it that way since I still want to keep py2 support for the moment